libsqlite3-sys/src/lib.rs

```diff
@@ -71,3 +71,64 @@ impl Default for sqlite3_vtab_cursor {
         unsafe { mem::zeroed() }
     }
 }
+
+#[cfg(all(target_arch = "wasm32", target_os = "unknown"))]
+mod allocator {
+    use std::alloc::{alloc, dealloc, realloc as rs_realloc, Layout};
+
+    #[no_mangle]
+    pub unsafe fn sqlite_malloc(len: usize) -> *mut u8 {
```

Comment:
These must be `extern "C"`.

I feel like we should either be overriding malloc, or should be using the normal config-based mechanism for overriding the allocator globally, or we should have a big comment explaining why we're doing it this way.